src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -237,6 +243,274 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
         this.dialogErrorSource.unsubscribe();
     }
 
+    getExerciseDetails() {
+        const exercise = this.programmingExercise;
+        return [
+            this.getExerciseDetailsGeneralSection(exercise),
+            this.getExerciseDetailsModeSection(exercise),
+            this.getExerciseDetailsLanguageSection(exercise),
+            this.getExerciseDetailsProblemSection(exercise),
+            this.getExerciseDetailsGradingSection(exercise),
+        ] as DetailOverviewSection[];
```

Comment:
Is a return type annotation not clearer?

Reply:
The casting here is needed because of the filtering of empty details, which is needed to remove the not shown details in some occasions. this works but is not perfectly beautiful 😅